core/elf.cc

```diff
@@ -372,9 +372,11 @@ void object::set_base(void* base)
         // Override the passed in value as the base for non-PICs (Position Dependant Executables)
         // needs to be set to 0 because all the addresses in it are absolute
         _base = 0x0;
+        _headers_start = reinterpret_cast<void*>(p->p_vaddr) + _ehdr.e_phoff;
     } else {
         // Otherwise for kernel, PIEs and shared libraries set the base as requested by caller
         _base = align(base, p->p_align, p->p_vaddr & (p->p_align - 1)) - p->p_vaddr;
+        _headers_start = _base + _ehdr.e_phoff;
```

Comment:
I couldn't find where you use this new _headers_start variable.
On the other hand, I see there are existing places in the code which use e_phoff, should they use this new variable, or not?